src/util.c

```diff
@@ -522,7 +522,7 @@ int string2ld(const char *s, size_t slen, long double *dp) {
     if (isspace(buf[0]) || eptr[0] != '\0' ||
         (size_t)(eptr-buf) != slen ||
         (errno == ERANGE &&
-            (value == HUGE_VAL || value == -HUGE_VAL || value == 0)) ||
+            (value == HUGE_VAL || value == -HUGE_VAL || fpclassify(value) == FP_ZERO)) ||
```

Comment:
let me see if i get this right.
you're saying that `strtold` could fail with `ERANGE`, setting `value` that's not 0 (`subnormal`?), but doing `==0` would still be true.
and in that case, we wanna fail `string2ld` rather than return that number as double to the caller rather than fail string2ld?

i guess this all depends on what are the usages for string2ld, and if they fall back to working with a string instead of a double, or do they return an error to the caller.

i'm guessing the number one usage is getDoubleFromObjectOrReply and getLongDoubleFromObjectOrReply, in which case maybe proceeding with 0 rather than error, is better. hence we prefer your fix.

however, looking at strtold docs it seems that `ERANGE` is only used together with positive or negative `HUGE_VAL`, which makes this code (and its fix) dead code.

Reply:
I see this was introduced here: 9d520a7f70d24ea663b8de9c7178624d76a8d846

the commit comment says:
```
    The problem was due to strtod() returning ERANGE in the following case
    specified by POSIX:
    
    "If the correct value would cause an underflow, a value whose magnitude
    is no greater than the smallest normalized positive number in the return
    type shall be returned and errno set to [ERANGE].".
    
    Now instead the returned value is accepted even when ERANGE is returned
    as long as the return value of the function is not negative or positive
    HUGE_VAL or zero.
```

but the example it contains doesn't seem to be affected either by this PR, or even by the complete removal of this condition.

Reply:
i've tested, and indeed `strtod` (not `strtold` which is used for zset scores nowadays), would return `ERANGE` for `9.9999999999999694e-310`, and both `==0` and `fpclassify` seem to indicate it's **not** zero (tested with `-O0` and `-O3` on my machine)

Reply:
> `==0`

Hi @oranagra , I confirm that with GCC I've seen the same behavior: both "fpclassify" and "==0" are working as expected. 
But with Intel compilers, ICC and ICX, where optimizations for 0 check are more aggressive, "==0" is true for mentioned functions, however should not be.